api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -194,6 +194,10 @@ type NodePoolSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Arch is immutable"
 	// +optional
 	Arch string `json:"arch,omitempty"`
+
+	// FailureDomain is the preferred failure domain for the NodePool.
```

Comment:
I would encourage expanding the description here.

Where does a user find out what the available and valid values are for this field for example?

Is it possible that a Nodepool could ever cover multiple failure domains?

What happens when the user doesn't specify a value here at all?